### PR TITLE
add spock functions to non-relic version

### PR DESCRIPTION
**Pull request by pattyshack**
This also refactor the bls errors into a non-tag specific file

This is needed to unblock #3696 (SPOCKProve is now used by the emulator intergration testing)

gh: #3695

### CODE REVIEW

**Review by SaveTheRbtz — APPROVED** [review]

**Review by tarakby — CHANGES_REQUESTED** [review]
There is no need to address the comments. I suggest https://github.com/onflow/flow-go/pull/3721 as a temporary fix to unblock your issue. 

Merging this PR will take more time as bumping the crypto version in flow-go requires more time (blocked by https://github.com/onflow/flow-go/pull/3441  and more work with Alex). 
I can get back to these changes and clean up `engine/execution/computer` later on.

**Pull request closed without merging**
